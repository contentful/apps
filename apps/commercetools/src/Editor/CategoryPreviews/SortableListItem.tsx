import React from "react";
import { SortableElement, SortableHandle } from "react-sortable-hoc";
import { css } from "emotion";
import {
  Card,
  CardDragHandle as FormaCardDragHandle,
  Heading,
  Icon,
  IconButton,
  Subheading,
  Tag,
  Typography
} from "@contentful/forma-36-react-components";
import tokens from "@contentful/forma-36-tokens";
import { Category } from "../../interfaces";

export interface Props {
  category: Category;
  disabled: boolean;
  onDelete: () => void;
  isSortable: boolean;
}

const styles = {
  card: css({
    display: "flex",
    padding: 0,
    position: "relative",
    ":not(:first-of-type)": css({
      marginTop: tokens.spacingXs
    })
  }),
  dragHandle: css({
    height: "auto"
  }),
  actions: css({
    position: "absolute",
    top: tokens.spacingXs,
    right: tokens.spacingXs,
    a: css({
      display: "inline-block",
      marginRight: tokens.spacingXs,
      svg: css({
        transition: `fill ${tokens.transitionDurationDefault} ${tokens.transitionEasingDefault}`
      }),
      "&:hover": {
        svg: css({
          fill: tokens.colorContrastDark
        })
      }
    })
  }),
  description: css({
    padding: tokens.spacingM,
    flex: "1 0 auto",
    display: "flex",
    flexDirection: "column",
    justifyContent: "center"
  }),
  name: (name?: string) =>
    css({
      fontSize: tokens.fontSizeL,
      marginBottom: tokens.spacing2Xs,
      ...(name && { textTransform: "capitalize" })
    }),
  slug: css({
    color: tokens.colorElementDarkest,
    fontSize: tokens.fontSizeS,
    marginBottom: 0
  })
};

const CardDragHandle = SortableHandle(() => (
  <FormaCardDragHandle className={styles.dragHandle}>
    Reorder category
  </FormaCardDragHandle>
));

export const SortableListItem = SortableElement<Props>(
  ({ category, disabled, isSortable, onDelete }: Props) => {
    const categoryIsMissing = !category.name;

    return (
      <Card className={styles.card}>
        <>
          {isSortable && <CardDragHandle />}
          <section className={styles.description}>
            <Typography>
              <Heading className={styles.name(category.name)}>
                {categoryIsMissing ? category.id : category.name}
              </Heading>
              {categoryIsMissing ? (
                <Tag tagType="negative">Category missing</Tag>
              ) : (
                <Subheading className={styles.slug}>{category.slug}</Subheading>
              )}
            </Typography>
          </section>
        </>
        {!disabled && (
          <div className={styles.actions}>
            {category.externalLink && (
              <a
                target="_blank"
                rel="noopener noreferrer"
                href={category.externalLink}
              >
                <Icon icon="ExternalLink" color="muted" />
              </a>
            )}
            <IconButton
              label="Delete"
              iconProps={{ icon: "Close" }}
              {...{
                buttonType: "muted",
                onClick: onDelete
              }}
            />
          </div>
        )}
      </Card>
    );
  }
);
