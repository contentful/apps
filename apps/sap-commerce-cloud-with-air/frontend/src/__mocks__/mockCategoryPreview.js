export const mockCategoryPreview = {
  sku: 'abc1234',
  image: '',
  id: '123',
  name: 'Mock Product',
  slug: '',
  isMissing: false,
};
