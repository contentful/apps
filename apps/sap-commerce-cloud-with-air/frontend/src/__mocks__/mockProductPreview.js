export const mockProductPreview = {
  sku: 'abc1234',
  image: '',
  id: '123',
  name: 'Mock Category',
  isMissing: false,
};
