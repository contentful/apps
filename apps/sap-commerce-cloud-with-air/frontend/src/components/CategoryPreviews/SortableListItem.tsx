import { ReactElement } from 'react';
import { SortableElement, SortableHandle } from 'react-sortable-hoc';
import { css } from 'emotion';
import { Badge, Card, Heading, IconButton, Subheading } from '@contentful/f36-components';
import { CloseIcon, ExternalLinkIcon } from '@contentful/f36-icons';
import tokens from '@contentful/forma-36-tokens';
import { Category } from '../../interfaces';

export interface Props {
  category: Category;
  disabled: boolean;
  onDelete: () => void;
  isSortable: boolean;
}

const styles = {
  card: css({
    display: 'flex',
    padding: 0,
    position: 'relative',
    ':not(:first-of-type)': css({
      marginTop: tokens.spacingXs,
    }),
  }),
  cardInner: css({
    display: 'flex',
  }),
  dragHandle: css({
    height: 'auto',
  }),
  actions: css({
    position: 'absolute',
    top: tokens.spacingXs,
    right: tokens.spacingXs,
    a: css({
      display: 'inline-block',
      marginRight: tokens.spacingXs,
      svg: css({
        transition: `fill ${tokens.transitionDurationDefault} ${tokens.transitionEasingDefault}`,
      }),
      '&:hover': {
        svg: css({
          fill: tokens.colorBlack,
        }),
      },
    }),
  }),
  description: css({
    padding: tokens.spacingM,
    flex: '1 0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
  }),
  heading: (category: Category) =>
    css({
      fontSize: tokens.fontSizeL,
      marginBottom: category.isMissing || !category.name ? 0 : tokens.spacing2Xs,
      ...(category.name && { textTransform: 'capitalize' }),
    }),
  subheading: css({
    color: tokens.gray500,
    fontSize: tokens.fontSizeS,
    marginBottom: 0,
  }),
  slug: css({
    color: tokens.colorBlack,
    fontSize: tokens.fontSizeS,
    marginBottom: 0,
  }),
};

const CardDragHandle = SortableHandle(({ drag }: { drag: ReactElement }) => <>{drag}</>);

function getCategoryIdentifier(category: Category) {
  return category.slug.length ? category.slug : category.id;
}

export const SortableListItem = SortableElement<Props>(
  ({ category, disabled, isSortable, onDelete }: Props) => {
    return (
      <Card
        className={styles.card}
        withDragHandle
        dragHandleRender={isSortable ? ({ drag }) => <CardDragHandle drag={drag} /> : undefined}>
        <div className={styles.cardInner}>
          <section className={styles.description}>
            <>
              <Heading className={styles.heading(category)}>
                {category.isMissing || !category.name
                  ? getCategoryIdentifier(category)
                  : category.name}
              </Heading>
              {category.isMissing && <Badge variant="negative">Category missing</Badge>}
              {!category.isMissing && category.name && (
                <Subheading className={styles.subheading}>
                  {getCategoryIdentifier(category)}
                </Subheading>
              )}
            </>
          </section>
        </div>
        {!disabled && (
          <div className={styles.actions}>
            {category.externalLink && (
              <a target="_blank" rel="noopener noreferrer" href={category.externalLink}>
                <ExternalLinkIcon color="muted" />
              </a>
            )}
            <IconButton
              aria-label="Delete"
              icon={<CloseIcon />}
              variant="transparent"
              onClick={onDelete}
            />
          </div>
        )}
      </Card>
    );
  },
);
