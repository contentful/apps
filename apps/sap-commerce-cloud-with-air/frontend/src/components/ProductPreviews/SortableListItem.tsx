import { useState, ReactElement } from 'react';
import { SortableElement, SortableHandle } from 'react-sortable-hoc';
import { css } from 'emotion';
import {
  Badge,
  Card,
  Heading,
  IconButton,
  SkeletonContainer,
  SkeletonImage,
  Subheading,
} from '@contentful/f36-components';
import { AssetIcon, CloseIcon, ErrorCircleIcon, ExternalLinkIcon } from '@contentful/f36-icons';
import tokens from '@contentful/forma-36-tokens';
import { Product } from '../../interfaces';

export interface Props {
  product: Product;
  disabled: boolean;
  onDelete: () => void;
  isSortable: boolean;
}

const IMAGE_SIZE = 48;

const styles = {
  card: css({
    padding: 0,
    position: 'relative',
    ':not(:first-of-type)': css({
      marginTop: tokens.spacingXs,
    }),
  }),
  cardInner: css({
    display: 'flex',
  }),
  imageWrapper: (imageHasLoaded: boolean) =>
    css({
      width: imageHasLoaded ? `${IMAGE_SIZE}px` : 0,
      height: imageHasLoaded ? `${IMAGE_SIZE}px` : 0,
      overflow: 'hidden',
      margin: imageHasLoaded ? tokens.spacingM : 0,
      position: 'relative',
      '> img': css({
        display: 'block',
        height: `${IMAGE_SIZE}px`,
        minWidth: 'auto',
        userSelect: 'none',
        position: 'absolute',
        left: '50%',
        top: '50%',
        transform: 'translate(-50%, -50%)',
      }),
    }),
  dragHandle: css({
    height: 'auto',
    borderBottomLeftRadius: tokens.borderRadiusMedium,
    borderTopLeftRadius: tokens.borderRadiusMedium,
    cursor: 'grab',
  }),
  actions: css({
    position: 'absolute',
    top: tokens.spacingXs,
    right: tokens.spacingXs,
    a: css({
      display: 'inline-block',
      marginRight: tokens.spacingXs,
      svg: css({
        transition: `fill ${tokens.transitionDurationDefault} ${tokens.transitionEasingDefault}`,
      }),
      '&:hover': {
        svg: css({
          fill: tokens.colorBlack,
        }),
      },
    }),
  }),
  description: css({
    flex: '1 0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
  }),
  heading: (product: Product) =>
    css({
      fontSize: tokens.fontSizeL,
      marginBottom: product.isMissing || !product.name ? 0 : tokens.spacing2Xs,
      ...(product.name && { textTransform: 'capitalize' }),
    }),
  subheading: css({
    color: tokens.gray500,
    fontSize: tokens.fontSizeS,
    marginBottom: 0,
  }),
  skeletonImage: css({
    width: `${IMAGE_SIZE}px`,
    height: `${IMAGE_SIZE}px`,
    padding: tokens.spacingM,
  }),
  errorImage: css({
    backgroundColor: tokens.gray100,
    borderRadius: '3px',
    margin: tokens.spacingM,
    width: `${IMAGE_SIZE}px`,
    height: `${IMAGE_SIZE}px`,
    position: 'relative',
    svg: css({
      fill: tokens.gray600,
      width: '100%',
      height: '50%',
      position: 'absolute',
      top: '50%',
      left: '50%',
      transform: 'translate(-50%, -50%)',
    }),
  }),
};

const CardDragHandle = SortableHandle(({ drag }: { drag: ReactElement }) => <>{drag}</>);

export const SortableListItem = SortableElement<Props>(
  ({ product, disabled, isSortable, onDelete }: Props) => {
    const [imageHasLoaded, setImageLoaded] = useState(false);
    const [imageHasErrored, setImageHasErrored] = useState(false);

    return (
      <Card
        className={styles.card}
        withDragHandle
        dragHandleRender={isSortable ? ({ drag }) => <CardDragHandle drag={drag} /> : undefined}>
        <div className={styles.cardInner}>
          {!imageHasLoaded && !imageHasErrored && (
            <SkeletonContainer className={styles.skeletonImage}>
              <SkeletonImage width={IMAGE_SIZE} height={IMAGE_SIZE} />
            </SkeletonContainer>
          )}
          {imageHasErrored && (
            <div className={styles.errorImage}>
              {product.isMissing ? (
                <ErrorCircleIcon testId="error-circle-icon" />
              ) : (
                <AssetIcon testId="asset-icon" />
              )}
            </div>
          )}
          {!imageHasErrored && (
            <div className={styles.imageWrapper(imageHasLoaded)}>
              <img
                style={{ display: imageHasLoaded ? 'block' : 'none' }}
                onLoad={() => setImageLoaded(true)}
                onError={() => setImageHasErrored(true)}
                src={product.image}
                alt={product.name}
                data-test-id="image"
              />
            </div>
          )}
          <section className={styles.description}>
            <Heading className={styles.heading(product)}>
              {product.isMissing || !product.name ? product.sku : product.name}
            </Heading>
            {product.isMissing && <Badge variant="negative">Product missing</Badge>}
            {!product.isMissing && product.name && (
              <Subheading className={styles.subheading}>{product.sku}</Subheading>
            )}
          </section>
        </div>
        {!disabled && (
          <div className={styles.actions}>
            {product.externalLink && (
              <a target="_blank" rel="noopener noreferrer" href={product.externalLink}>
                <ExternalLinkIcon color="muted" />
              </a>
            )}
            <IconButton
              aria-label="Delete"
              icon={<CloseIcon />}
              variant="transparent"
              onClick={onDelete}
            />
          </div>
        )}
      </Card>
    );
  },
);
