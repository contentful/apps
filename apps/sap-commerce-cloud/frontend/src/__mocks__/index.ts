export { makeSdkMock } from './mockSdk';
export { mockProductPreview } from './mockProductPreview';
