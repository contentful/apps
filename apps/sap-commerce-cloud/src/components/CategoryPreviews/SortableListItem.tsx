import React from 'react';
import { SortableElement, SortableHandle } from 'react-sortable-hoc';
import { css } from 'emotion';
import {
  Card,
  CardDragHandle as FormaCardDragHandle,
  Heading,
  Icon,
  IconButton,
  Subheading,
  Tag,
  Typography,
} from '@contentful/forma-36-react-components';
import tokens from '@contentful/forma-36-tokens';
import { Category } from '../../interfaces';

export interface Props {
  category: Category;
  disabled: boolean;
  onDelete: () => void;
  isSortable: boolean;
}

const styles = {
  card: css({
    display: 'flex',
    padding: 0,
    position: 'relative',
    ':not(:first-of-type)': css({
      marginTop: tokens.spacingXs,
    }),
  }),
  dragHandle: css({
    height: 'auto',
  }),
  actions: css({
    position: 'absolute',
    top: tokens.spacingXs,
    right: tokens.spacingXs,
    a: css({
      display: 'inline-block',
      marginRight: tokens.spacingXs,
      svg: css({
        transition: `fill ${tokens.transitionDurationDefault} ${tokens.transitionEasingDefault}`,
      }),
      '&:hover': {
        svg: css({
          fill: tokens.colorContrastDark,
        }),
      },
    }),
  }),
  description: css({
    padding: tokens.spacingM,
    flex: '1 0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
  }),
  heading: (category: Category) =>
    css({
      fontSize: tokens.fontSizeL,
      marginBottom: category.isMissing || !category.name ? 0 : tokens.spacing2Xs,
      ...(category.name && { textTransform: 'capitalize' }),
    }),
  subheading: css({
    color: tokens.colorElementDarkest,
    fontSize: tokens.fontSizeS,
    marginBottom: 0,
  }),
  slug: css({
    color: tokens.colorElementDarkest,
    fontSize: tokens.fontSizeS,
    marginBottom: 0,
  }),
};

const CardDragHandle = SortableHandle(() => (
  <FormaCardDragHandle className={styles.dragHandle}>Reorder category</FormaCardDragHandle>
));

function getCategoryIdentifier(category: Category) {
  return category.slug.length ? category.slug : category.id;
}

export const SortableListItem = SortableElement<Props>(
  ({ category, disabled, isSortable, onDelete }: Props) => {
    return (
      <Card className={styles.card}>
        <>
          {isSortable && <CardDragHandle />}
          <section className={styles.description}>
            <Typography>
              <Heading className={styles.heading(category)}>
                {category.isMissing || !category.name
                  ? getCategoryIdentifier(category)
                  : category.name}
              </Heading>
              {category.isMissing && <Tag tagType="negative">Category missing</Tag>}
              {!category.isMissing && category.name && (
                <Subheading className={styles.subheading}>
                  {getCategoryIdentifier(category)}
                </Subheading>
              )}
            </Typography>
          </section>
        </>
        {!disabled && (
          <div className={styles.actions}>
            {category.externalLink && (
              <a target="_blank" rel="noopener noreferrer" href={category.externalLink}>
                <Icon icon="ExternalLink" color="muted" />
              </a>
            )}
            <IconButton
              label="Delete"
              iconProps={{ icon: 'Close' }}
              buttonType="muted"
              onClick={onDelete}
            />
          </div>
        )}
      </Card>
    );
  }
);
