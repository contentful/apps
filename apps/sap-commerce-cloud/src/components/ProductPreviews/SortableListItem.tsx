import { FC, useState } from 'react';
import { useSortable } from '@dnd-kit/sortable';
import { CSS } from '@dnd-kit/utilities';
import { Product } from '../../interfaces';
import {
  AssetIcon,
  Badge,
  Card,
  Flex,
  Heading,
  IconButton,
  SkeletonContainer,
  SkeletonImage,
  Subheading,
} from '@contentful/f36-components';
import tokens from '@contentful/f36-tokens';
import { styles } from './SortableListItem.styles';
import { CloseIcon, ErrorCircleIcon, ExternalLinkIcon } from '@contentful/f36-icons';

export interface Props {
  product: Product;
  disabled: boolean;
  onDelete: () => void;
  isSortable: boolean;
}

const IMAGE_SIZE = 48;

export const SortableListItem: FC<Props> = ({ product, disabled, onDelete, isSortable }: Props) => {
  const [imageHasLoaded, setImageLoaded] = useState(false);
  const [imageHasErrored, setImageHasErrored] = useState(false);

  const { attributes, listeners, setNodeRef, transform, transition } = useSortable({
    id: product.productUrl,
  });

  const style = isSortable
    ? {
        transform: CSS.Transform.toString(transform),
        transition,
        marginTop: tokens.spacingS,
      }
    : { marginTop: tokens.spacingS };

  return (
    <div
      ref={setNodeRef}
      style={style}
      key={product.id}
      {...attributes}
      {...listeners}
      data-testid={`sortable-item-${product.id}`}>
      <Card withDragHandle={isSortable} className={styles.card} data-testid="sortable-list-item">
        <Flex>
          {!imageHasLoaded && !imageHasErrored && (
            <SkeletonContainer className={styles.skeletonImage}>
              <SkeletonImage width={IMAGE_SIZE} height={IMAGE_SIZE} />
            </SkeletonContainer>
          )}
          {imageHasErrored && (
            <div className={styles.errorImage}>
              {product.isMissing ? (
                <ErrorCircleIcon data-testid="icon" />
              ) : (
                <AssetIcon data-testid="icon" />
              )}
            </div>
          )}
          {!imageHasErrored && (
            <div className={styles.imageWrapper(imageHasLoaded)}>
              <img
                style={{ display: imageHasLoaded ? 'block' : 'none' }}
                onLoad={() => setImageLoaded(true)}
                onError={() => setImageHasErrored(true)}
                src={product.image}
                alt={product.name}
                data-testid="image"
              />
            </div>
          )}
          <section className={styles.description}>
            <Heading className={styles.heading(product)}>
              {product.isMissing || !product.name ? product.sku : product.name}
            </Heading>
            {product.isMissing && <Badge variant="negative">Product missing</Badge>}
            {!product.isMissing && product.name && (
              <Subheading className={styles.subheading}>{product.sku}</Subheading>
            )}
          </section>
          {!disabled && (
            <div className={styles.actions}>
              {product.externalLink && (
                <a target="_blank" rel="noopener noreferrer" href={product.externalLink}>
                  <ExternalLinkIcon variant="muted" />
                </a>
              )}
              <IconButton
                icon={<CloseIcon variant="muted" />}
                aria-label="Delete"
                onClick={onDelete}
              />
            </div>
          )}
        </Flex>
      </Card>
    </div>
  );
};
