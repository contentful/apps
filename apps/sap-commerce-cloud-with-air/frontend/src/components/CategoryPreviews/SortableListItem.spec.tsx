import { configure, render, cleanup } from '@testing-library/react';
import { Props, SortableListItem } from './SortableListItem';
import { mockCategoryPreview } from '../../__mocks__/mockCategoryPreview';

configure({
  testIdAttribute: 'data-test-id',
});

const defaultProps: Props = {
  category: mockCategoryPreview,
  disabled: false,
  onDelete: jest.fn(),
  isSortable: false,
};

jest.mock('react-sortable-hoc');

const renderComponent = (props: Props) => {
  return render(<SortableListItem index={0} {...props} />);
};

describe('SortableListItem', () => {
  afterEach(cleanup);

  it('should render successfully', async () => {
    const component = renderComponent(defaultProps);
    expect(component.container).toMatchSnapshot();
  });

  it('should render successfully the error variation for missing slug', () => {
    const component = renderComponent({
      ...defaultProps,
      category: { ...mockCategoryPreview, name: '' },
    });
    expect(component.container).toMatchSnapshot();
  });
});
