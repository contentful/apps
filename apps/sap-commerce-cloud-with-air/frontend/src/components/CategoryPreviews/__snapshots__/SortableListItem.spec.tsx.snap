// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortableListItem should render successfully 1`] = `
<div>
  <article
    class="css-jg0czf"
    data-test-id="cf-ui-card"
  >
    <div
      class="css-11p49tw"
      data-test-id="cf-ui-drag-handle"
      role="button"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="css-en3na7"
        data-test-id="cf-ui-icon"
        role="img"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
      <span
        class="css-1b512s9"
      >
        Reorder entry
      </span>
    </div>
    <div
      class="css-1cjnmwb"
      data-card-part="wrapper"
    >
      <div
        class="css-161cbg"
        data-card-part="content"
      >
        <div
          class="css-k008qs"
        >
          <section
            class="css-ijlwe4"
          >
            <h1
              class="css-19gqs9u"
              data-test-id="cf-ui-heading"
            >
              Mock Product
            </h1>
            <h3
              class="css-1dno5oq"
              data-test-id="cf-ui-subheading"
            >
              123
            </h3>
          </section>
        </div>
        <div
          class="css-mfolfb"
        >
          <button
            aria-label="Delete"
            class="css-zob9g7"
            data-test-id="cf-ui-icon-button"
            type="button"
          >
            <span
              class="css-k008qs"
            >
              <svg
                aria-hidden="true"
                class="css-1jrff5x"
                data-test-id="cf-ui-icon"
                role="img"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </article>
</div>
`;

exports[`SortableListItem should render successfully the error variation for missing slug 1`] = `
<div>
  <article
    class="css-jg0czf"
    data-test-id="cf-ui-card"
  >
    <div
      class="css-11p49tw"
      data-test-id="cf-ui-drag-handle"
      role="button"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="css-en3na7"
        data-test-id="cf-ui-icon"
        role="img"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
      <span
        class="css-1b512s9"
      >
        Reorder entry
      </span>
    </div>
    <div
      class="css-1cjnmwb"
      data-card-part="wrapper"
    >
      <div
        class="css-161cbg"
        data-card-part="content"
      >
        <div
          class="css-k008qs"
        >
          <section
            class="css-ijlwe4"
          >
            <h1
              class="css-1t11k68"
              data-test-id="cf-ui-heading"
            >
              123
            </h1>
            
          </section>
        </div>
        <div
          class="css-mfolfb"
        >
          <button
            aria-label="Delete"
            class="css-zob9g7"
            data-test-id="cf-ui-icon-button"
            type="button"
          >
            <span
              class="css-k008qs"
            >
              <svg
                aria-hidden="true"
                class="css-1jrff5x"
                data-test-id="cf-ui-icon"
                role="img"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </article>
</div>
`;
